applications/integrationtests/mechanics/GradientDamage.cpp

```diff
@@ -157,23 +162,63 @@ BOOST_AUTO_TEST_CASE(Integrand2D)
     DofType d("Displacements", 2);
     ScalarDofType eeq("NonlocalEquivalentStrains");
 
-    double c = 1.;
+    double c = 0.1;
     using Gdm = Integrands::GradientDamage<2, Constitutive::DamageLawExponential>;
+    using Neumann = Integrands::NeumannBc<2>;
+
     Gdm gdm(d, eeq, c, elasticLaw, dmg, strainNorm);
-    gdm.mKappas.resize(1, 1);
+    Neumann neumann(d, Eigen::Vector2d(.42, .12));
 
-    MeshFem mesh = UnitMeshFem::CreateQuads(1, 1);
-    AddDofInterpolation(&mesh, d);
-    AddDofInterpolation(&mesh, eeq);
+    MeshGmsh gmsh(meshFile);
+
+    auto& mesh = gmsh.GetMeshFEM();
+    auto& matrixElements = gmsh.GetPhysicalGroup("matrix");
+    auto& leftElements = gmsh.GetPhysicalGroup("left");
+    AddDofInterpolation(&mesh, d, matrixElements);
+    AddDofInterpolation(&mesh, d, leftElements);
+    AddDofInterpolation(&mesh, eeq, matrixElements);
+    AddDofInterpolation(&mesh, eeq, leftElements);
 
-    auto& element = mesh.Elements[0];
-    Eigen::Vector2d ipCoords(0, 0);
-    Jacobian jacobian(element.CoordinateElement().ExtractNodeValues(),
-                      element.CoordinateElement().GetDerivativeShapeFunctions(ipCoords),
-                      element.CoordinateElement().GetDofDimension());
-    CellData cd(element, 0);
-    CellIpData cipd(cd, jacobian, ipCoords, 0);
+    IntegrationType2D3NGauss12Ip integration;
+    CellStorage cellStorage;
+    auto cells = cellStorage.AddCells(matrixElements, integration);
+    auto cellsLeft = cellStorage.AddCells(leftElements, integration);
+    gdm.mKappas.setZero(cells.Size(), integration.GetNumIntegrationPoints());
+
+    TimeDependentProblem equations(&mesh);
+
+    equations.AddHessian0Function(cells, TimeDependentProblem::Bind(gdm, &Gdm::Hessian0));
+    equations.AddGradientFunction(cells, TimeDependentProblem::Bind(gdm, &Gdm::Gradient));
+    equations.AddUpdateFunction(cells, TimeDependentProblem::Bind(gdm, &Gdm::Update));
+    equations.AddGradientFunction(cellsLeft, TimeDependentProblem::Bind(neumann, &Neumann::ExternalLoad));
+
+    QuasistaticSolver problem(equations, {d, eeq});
+
+    using namespace Constraint;
+    Constraints constraints;
+    constraints.Add(d, Component(mesh.NodeAtCoordinate(Eigen::Vector2d::Zero(), d), {eDirection::X}));
+    constraints.Add(d, Component(mesh.NodesAtAxis(eDirection::Y, d), {eDirection::Y}));
+    constraints.Add(d, Component(mesh.NodesAtAxis(eDirection::Y, d, 16), {eDirection::Y}, RhsRamp(1, 0.01)));
+
+    problem.SetConstraints(constraints);
+    problem.mTolerance = 1.e-6;
+
+
+    using namespace NuTo::Visualize;
+    PostProcess visu("./GDMout");
+    visu.DefineVisualizer("GDM", cells, VoronoiHandler(VoronoiGeometryTriangle(integration)));
+    visu.Add("GDM", d);
+    visu.Add("GDM", eeq);
+    visu.Add("GDM",
+             [&](const NuTo::CellIpData& data) {
+                 return EigenCompanion::ToEigen(gdm.mDamageLaw.Damage(gdm.Kappa(data)));
```

Comment:
Maybe we should overload the `Add`/`Visualizer::CellData` functions for `double`s, so that you don't need the `ToEigen` as a user.

Reply:
Good point. Do you think it is enough to do this overload in `PostProcess`?